config/v1/types_securityprofile.go

```diff
@@ -0,0 +1,333 @@
+	// the minimum version is determined by the security profile type.
+	//
+	//   SecurityProfileType Modern:       SecurityProtocolTLS12Version
+	//   SecurityProfileType Intermediate: SecurityProtocolTLS10Version
+	//   SecurityProfileType Old:          SecurityProtocolTLS10Version
+	//
+	// Supported minimum versions are:
+	//
+	//   "TLSv1.3": Version 1.3 of the TLS security protocol used for securing network connections.
+	//   "TLSv1.2": Version 1.2 of the TLS security protocol used for securing network connections.
+	//   "TLSv1.1": Version 1.1 of the TLS security protocol used for securing network connections.
+	//   "TLSv1.0": Version 1.0 of the TLS security protocol used for securing network connections.
+	//
+	// +optional
+	MinimumVersion *SecurityProtocolVersion `json:"minimumVersion,omitempty"`
+	// maximumVersion enforces use of the specified SecurityProtocolVersion or older
+	// on SSL connections. maximumVersion must be higher than or equal to minimumVersion.
+	//
+	// If unset and minimumVersion is set, maximumVersion will be set
+	// to minimumVersion. If minimumVersion and maximumVersion are unset,
+	// the maximum version is determined by the security profile type.
+	//
+	//   SecurityProfileType Modern:       SecurityProtocolTLS12Version
+	//   SecurityProfileType Intermediate: SecurityProtocolTLS12Version
+	//   SecurityProfileType Old:          SecurityProtocolTLS12Version
+	//
+	// Supported maximum versions are the same as minimum versions.
+	//
+	// +optional
+	MaximumVersion *SecurityProtocolVersion `json:"maximumVersion,omitempty"`
+}
+
+// SecurityProtocolVersion is a way to specify the TLS security protocol version used to
+// secure network connections.
+type SecurityProtocolVersion string
+
+const (
+	// TLSv1.0 is version 1.0 of the TLS security protocol used for securing network connections.
+	SecurityProtocolTLS10Version SecurityProtocolVersion = "TLSv1.0"
+	// TLSv1.1 is version 1.1 of the TLS security protocol used for securing network connections.
+	SecurityProtocolTLS11Version SecurityProtocolVersion = "TLSv1.1"
+	// TLSv1.2 is version 1.2 of the TLS security protocol used for securing network connections.
+	SecurityProtocolTLS12Version SecurityProtocolVersion = "TLSv1.2"
+	// TLSv1.3 is version 1.3 of the TLS security protocol used for securing network connections.
+	SecurityProtocolTLS13Version SecurityProtocolVersion = "TLSv1.3"
+)
+
+// DHParamSize sets the maximum size of the Diffie-Hellman parameters used for
+// generating the ephemeral/temporary Diffie-Hellman key.
+type DHParamSize string
+
+const (
+	// 1024 is a Diffie-Hellman parameter of 1024 bits.
+	DHParamSize1024 DHParamSize = "1024"
+	// 2048 is a Diffie-Hellman parameter of 2048 bits.
+	DHParamSize2048 DHParamSize = "2048"
+)
+
+// SecurityProfileStatus defines the observed status of the SecurityProfile.
+type SecurityProfileStatus struct {
+	// SecurityProfileSettings is the actual settings of the security profile in use.
```

Comment:
Godoc should use lower-cased names: `SecurityProfileSettings`→`securityProfileSettings`.